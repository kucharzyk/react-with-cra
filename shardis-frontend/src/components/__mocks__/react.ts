import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
