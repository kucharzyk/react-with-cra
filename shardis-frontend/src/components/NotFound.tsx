import * as React from 'react';
import { StatelessComponent } from 'react';

const NotFound: StatelessComponent = () => (
  <div className="NotFound">
    <h2>This is NotFound component</h2>
  </div>
);

export default NotFound;
