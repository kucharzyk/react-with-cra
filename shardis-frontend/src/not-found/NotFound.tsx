import * as React from 'react';
import { StatelessComponent } from 'react';
import { Col, Row } from 'antd';

const NotFound: StatelessComponent = () => (
  <div className="NotFound">
    <h2>This is NotFound component</h2>
    <Row>
      <Col xs={0} sm={4} md={6} lg={8} xl={10}>Col1</Col>
      <Col xs={24} sm={16} md={12} lg={8} xl={4}>Col2</Col>
      <Col xs={0} sm={4} md={6} lg={8} xl={10}>Col3</Col>
    </Row>

    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Donec ullamcorper ultrices mi, hendrerit maximus metus.
      Morbi gravida imperdiet sapien iaculis tincidunt.
      Vivamus vehicula aliquet
      mauris, non gravida felis rhoncus a.
      Nam in leo nec nisi eleifend auctor.
      Vivamus imperdiet elementum congue.
      Sed in mollis felis, vel lacinia quam.
      Donec lacinia, arcu in iaculis gravida, ipsum
      justo consequat enim, non blandit ante risus at augue.
      Nam id nulla magna.
      Maecenas consectetur lectus ac eleifend lobortis.
      Vestibulum finibus pretium urna luctus euismod.
      Sed vehicula massa et
      neque fermentum, sit amet faucibus dui interdum.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Maecenas sit amet mi dignissim sem euismod faucibus eu
      quis ex.

      Aenean malesuada mollis nulla, sed facilisis felis elementum ut.
      Sed accumsan posuere elementum.
      Proin faucibus enim mi, ac vulputate urna elementum id.
      Sed in suscipit turpis.
      Nam id massa non
      ligula molestie bibendum.
      Duis convallis nibh lorem, quis egestas mi sagittis non.
      Integer semper porttitor porta.
      Suspendisse potenti.
      Curabitur non felis ut risus consequat pharetra id
      vulputate sem.
      Vestibulum id nunc nisi.

      Nam vestibulum sem vel leo lacinia porta.
      Ut enim nulla, lobortis eu laoreet eget, consectetur quis risus.
      Quisque sed lacus nec nunc sodales facilisis nec et sapien.
      Maecenas at nisl pretium,
      vestibulum nulla in, rhoncus orci.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Nulla tristique in felis sed aliquam.
      Vestibulum iaculis vel metus at feugiat.
      Nulla suscipit,
      tellus a semper elementum, lorem elit varius turpis, sed tempor urna neque vel sem.
      Ut vel fringilla eros.
      Ut leo turpis, molestie a lacus eu, ultrices varius sapien.
      Etiam eu dolor a lacus
      laoreet tempus.
      Mauris rutrum dui ac tellus fringilla, sit amet consequat sapien aliquet.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Nam a sagittis
      diam.

      In hac habitasse platea dictumst.
      Duis mollis vitae risus a hendrerit.
      Integer venenatis, diam ac mollis consectetur, dolor diam rutrum tortor, eu suscipit lectus ligula a libero.
      Suspendisse
      sit amet sem lacinia, pellentesque massa et, rhoncus mauris.
      Morbi suscipit, nibh quis interdum tincidunt, libero purus lacinia ipsum, in interdum diam lacus et nulla.
      Mauris non semper eros.

      Mauris rutrum tincidunt purus.
      Curabitur fermentum at odio non sagittis.
      Quisque commodo dolor eros.
      Donec eget vehicula quam, non mattis augue.
      Vivamus egestas, massa sed mollis efficitur, nisi
      felis laoreet ante, id tincidunt nibh enim non dolor.

      Nam non leo tristique, dapibus nisi at, iaculis lectus.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Cras rutrum, lectus nec imperdiet molestie, est orci varius tellus, id
      gravida massa mi id dui.
      Ut finibus massa et nisl viverra, tempus tempus mauris bibendum.
      Nam ac dignissim urna, ornare tincidunt lorem.
      Nulla aliquam tempor sagittis.
      Vestibulum iaculis leo
      hendrerit felis tristique accumsan.
      Quisque ut magna hendrerit, consectetur nisl et, auctor neque.
      Fusce facilisis tellus vitae ipsum malesuada rutrum.

      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Donec ullamcorper ultrices mi, hendrerit maximus metus.
      Morbi gravida imperdiet sapien iaculis tincidunt.
      Vivamus vehicula aliquet
      mauris, non gravida felis rhoncus a.
      Nam in leo nec nisi eleifend auctor.
      Vivamus imperdiet elementum congue.
      Sed in mollis felis, vel lacinia quam.
      Donec lacinia, arcu in iaculis gravida, ipsum
      justo consequat enim, non blandit ante risus at augue.
      Nam id nulla magna.
      Maecenas consectetur lectus ac eleifend lobortis.
      Vestibulum finibus pretium urna luctus euismod.
      Sed vehicula massa et
      neque fermentum, sit amet faucibus dui interdum.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Maecenas sit amet mi dignissim sem euismod faucibus eu
      quis ex.

      Aenean malesuada mollis nulla, sed facilisis felis elementum ut.
      Sed accumsan posuere elementum.
      Proin faucibus enim mi, ac vulputate urna elementum id.
      Sed in suscipit turpis.
      Nam id massa non
      ligula molestie bibendum.
      Duis convallis nibh lorem, quis egestas mi sagittis non.
      Integer semper porttitor porta.
      Suspendisse potenti.
      Curabitur non felis ut risus consequat pharetra id
      vulputate sem.
      Vestibulum id nunc nisi.

      Nam vestibulum sem vel leo lacinia porta.
      Ut enim nulla, lobortis eu laoreet eget, consectetur quis risus.
      Quisque sed lacus nec nunc sodales facilisis nec et sapien.
      Maecenas at nisl pretium,
      vestibulum nulla in, rhoncus orci.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Nulla tristique in felis sed aliquam.
      Vestibulum iaculis vel metus at feugiat.
      Nulla suscipit,
      tellus a semper elementum, lorem elit varius turpis, sed tempor urna neque vel sem.
      Ut vel fringilla eros.
      Ut leo turpis, molestie a lacus eu, ultrices varius sapien.
      Etiam eu dolor a lacus
      laoreet tempus.
      Mauris rutrum dui ac tellus fringilla, sit amet consequat sapien aliquet.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Nam a sagittis
      diam.

      In hac habitasse platea dictumst.
      Duis mollis vitae risus a hendrerit.
      Integer venenatis, diam ac mollis consectetur, dolor diam rutrum tortor, eu suscipit lectus ligula a libero.
      Suspendisse
      sit amet sem lacinia, pellentesque massa et, rhoncus mauris.
      Morbi suscipit, nibh quis interdum tincidunt, libero purus lacinia ipsum, in interdum diam lacus et nulla.
      Mauris non semper eros.

      Mauris rutrum tincidunt purus.
      Curabitur fermentum at odio non sagittis.
      Quisque commodo dolor eros.
      Donec eget vehicula quam, non mattis augue.
      Vivamus egestas, massa sed mollis efficitur, nisi
      felis laoreet ante, id tincidunt nibh enim non dolor.

      Nam non leo tristique, dapibus nisi at, iaculis lectus.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Cras rutrum, lectus nec imperdiet molestie, est orci varius tellus, id
      gravida massa mi id dui.
      Ut finibus massa et nisl viverra, tempus tempus mauris bibendum.
      Nam ac dignissim urna, ornare tincidunt lorem.
      Nulla aliquam tempor sagittis.
      Vestibulum iaculis leo
      hendrerit felis tristique accumsan.
      Quisque ut magna hendrerit, consectetur nisl et, auctor neque.
      Fusce facilisis tellus vitae ipsum malesuada rutrum.

      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Donec ullamcorper ultrices mi, hendrerit maximus metus.
      Morbi gravida imperdiet sapien iaculis tincidunt.
      Vivamus vehicula aliquet
      mauris, non gravida felis rhoncus a.
      Nam in leo nec nisi eleifend auctor.
      Vivamus imperdiet elementum congue.
      Sed in mollis felis, vel lacinia quam.
      Donec lacinia, arcu in iaculis gravida, ipsum
      justo consequat enim, non blandit ante risus at augue.
      Nam id nulla magna.
      Maecenas consectetur lectus ac eleifend lobortis.
      Vestibulum finibus pretium urna luctus euismod.
      Sed vehicula massa et
      neque fermentum, sit amet faucibus dui interdum.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Maecenas sit amet mi dignissim sem euismod faucibus eu
      quis ex.

      Aenean malesuada mollis nulla, sed facilisis felis elementum ut.
      Sed accumsan posuere elementum.
      Proin faucibus enim mi, ac vulputate urna elementum id.
      Sed in suscipit turpis.
      Nam id massa non
      ligula molestie bibendum.
      Duis convallis nibh lorem, quis egestas mi sagittis non.
      Integer semper porttitor porta.
      Suspendisse potenti.
      Curabitur non felis ut risus consequat pharetra id
      vulputate sem.
      Vestibulum id nunc nisi.

      Nam vestibulum sem vel leo lacinia porta.
      Ut enim nulla, lobortis eu laoreet eget, consectetur quis risus.
      Quisque sed lacus nec nunc sodales facilisis nec et sapien.
      Maecenas at nisl pretium,
      vestibulum nulla in, rhoncus orci.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Nulla tristique in felis sed aliquam.
      Vestibulum iaculis vel metus at feugiat.
      Nulla suscipit,
      tellus a semper elementum, lorem elit varius turpis, sed tempor urna neque vel sem.
      Ut vel fringilla eros.
      Ut leo turpis, molestie a lacus eu, ultrices varius sapien.
      Etiam eu dolor a lacus
      laoreet tempus.
      Mauris rutrum dui ac tellus fringilla, sit amet consequat sapien aliquet.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Nam a sagittis
      diam.

      In hac habitasse platea dictumst.
      Duis mollis vitae risus a hendrerit.
      Integer venenatis, diam ac mollis consectetur, dolor diam rutrum tortor, eu suscipit lectus ligula a libero.
      Suspendisse
      sit amet sem lacinia, pellentesque massa et, rhoncus mauris.
      Morbi suscipit, nibh quis interdum tincidunt, libero purus lacinia ipsum, in interdum diam lacus et nulla.
      Mauris non semper eros.

      Mauris rutrum tincidunt purus.
      Curabitur fermentum at odio non sagittis.
      Quisque commodo dolor eros.
      Donec eget vehicula quam, non mattis augue.
      Vivamus egestas, massa sed mollis efficitur, nisi
      felis laoreet ante, id tincidunt nibh enim non dolor.

      Nam non leo tristique, dapibus nisi at, iaculis lectus.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Cras rutrum, lectus nec imperdiet molestie, est orci varius tellus, id
      gravida massa mi id dui.
      Ut finibus massa et nisl viverra, tempus tempus mauris bibendum.
      Nam ac dignissim urna, ornare tincidunt lorem.
      Nulla aliquam tempor sagittis.
      Vestibulum iaculis leo
      hendrerit felis tristique accumsan.
      Quisque ut magna hendrerit, consectetur nisl et, auctor neque.
      Fusce facilisis tellus vitae ipsum malesuada rutrum.

      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Donec ullamcorper ultrices mi, hendrerit maximus metus.
      Morbi gravida imperdiet sapien iaculis tincidunt.
      Vivamus vehicula aliquet
      mauris, non gravida felis rhoncus a.
      Nam in leo nec nisi eleifend auctor.
      Vivamus imperdiet elementum congue.
      Sed in mollis felis, vel lacinia quam.
      Donec lacinia, arcu in iaculis gravida, ipsum
      justo consequat enim, non blandit ante risus at augue.
      Nam id nulla magna.
      Maecenas consectetur lectus ac eleifend lobortis.
      Vestibulum finibus pretium urna luctus euismod.
      Sed vehicula massa et
      neque fermentum, sit amet faucibus dui interdum.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Maecenas sit amet mi dignissim sem euismod faucibus eu
      quis ex.

      Aenean malesuada mollis nulla, sed facilisis felis elementum ut.
      Sed accumsan posuere elementum.
      Proin faucibus enim mi, ac vulputate urna elementum id.
      Sed in suscipit turpis.
      Nam id massa non
      ligula molestie bibendum.
      Duis convallis nibh lorem, quis egestas mi sagittis non.
      Integer semper porttitor porta.
      Suspendisse potenti.
      Curabitur non felis ut risus consequat pharetra id
      vulputate sem.
      Vestibulum id nunc nisi.

      Nam vestibulum sem vel leo lacinia porta.
      Ut enim nulla, lobortis eu laoreet eget, consectetur quis risus.
      Quisque sed lacus nec nunc sodales facilisis nec et sapien.
      Maecenas at nisl pretium,
      vestibulum nulla in, rhoncus orci.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Nulla tristique in felis sed aliquam.
      Vestibulum iaculis vel metus at feugiat.
      Nulla suscipit,
      tellus a semper elementum, lorem elit varius turpis, sed tempor urna neque vel sem.
      Ut vel fringilla eros.
      Ut leo turpis, molestie a lacus eu, ultrices varius sapien.
      Etiam eu dolor a lacus
      laoreet tempus.
      Mauris rutrum dui ac tellus fringilla, sit amet consequat sapien aliquet.
      Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
      Nam a sagittis
      diam.

      In hac habitasse platea dictumst.
      Duis mollis vitae risus a hendrerit.
      Integer venenatis, diam ac mollis consectetur, dolor diam rutrum tortor, eu suscipit lectus ligula a libero.
      Suspendisse
      sit amet sem lacinia, pellentesque massa et, rhoncus mauris.
      Morbi suscipit, nibh quis interdum tincidunt, libero purus lacinia ipsum, in interdum diam lacus et nulla.
      Mauris non semper eros.

      Mauris rutrum tincidunt purus.
      Curabitur fermentum at odio non sagittis.
      Quisque commodo dolor eros.
      Donec eget vehicula quam, non mattis augue.
      Vivamus egestas, massa sed mollis efficitur, nisi
      felis laoreet ante, id tincidunt nibh enim non dolor.

      Nam non leo tristique, dapibus nisi at, iaculis lectus.
      Interdum et malesuada fames ac ante ipsum primis in faucibus.
      Cras rutrum, lectus nec imperdiet molestie, est orci varius tellus, id
      gravida massa mi id dui.
      Ut finibus massa et nisl viverra, tempus tempus mauris bibendum.
      Nam ac dignissim urna, ornare tincidunt lorem.
      Nulla aliquam tempor sagittis.
      Vestibulum iaculis leo
      hendrerit felis tristique accumsan.
      Quisque ut magna hendrerit, consectetur nisl et, auctor neque.
      Fusce facilisis tellus vitae ipsum malesuada rutrum.
    </p>
  </div>
);

export default NotFound;
