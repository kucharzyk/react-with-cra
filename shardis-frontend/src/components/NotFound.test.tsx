import * as React from 'react';
import * as ReactDOM from 'react-dom';

import NotFound from './NotFound';
import { create } from 'react-test-renderer';

describe('NotFound', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NotFound/>, div);
  });

  it('renders correctly', () => {
    const tree = create(<NotFound/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
