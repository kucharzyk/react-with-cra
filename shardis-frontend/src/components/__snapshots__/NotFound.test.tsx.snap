// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
<div
  className="NotFound"
>
  <h2>
    This is NotFound component
  </h2>
</div>
`;
